### PR TITLE
CI: add clang-python2 to Actions, remove duplicates from Travis

### PR DESCRIPTION
**Pull request by alltilla**
We have the `clang` jobs in Travis, only with `python2`, so it is enough to add these tests for now, but the `python3` tests will come later, too.

I have also removed the duplicated jobs in Travis with this PR.

No doc or news entry needed, IMO.

Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by alltilla** [comment]
Note: There is one problem still, with the GitHub Actions CI: `make python-pylint` fails, but we ignore it.
It only checks `debuggercli`, which as far as I know, is an unfinished module. It might be worth to completely turn off the checks for it.

**Comment by bazsi** [comment]
Why would it be unfinished?

File: .travis.yml [hunk 1]
```diff
@@ -1,194 +1,20 @@
...
-            ..
-        - ninja all check install
-
     - env: B=cmake-arm
       compiler: gcc
       sudo: required
       arch: arm64
       addons:
         apt:
+          sources:
```
Comment (Kokan) [inline .travis.yml:15]: IIRC every package listed below actually available in the default repository, and so it is not needed to include this repository. Specially due to this specific repo url do not provide arm based package, that supplied via raspberry path.

**Comment by MrAnno** [comment]
Please don't forget a clang job with the `enable-force-gnu99` configure flag enabled. It helps catching errors when older gcc versions are used.

File: .travis.yml (hunk 1 above)
Comment (alltilla) [reply .travis.yml]: Thanks, fixed :)

**Comment by alltilla** [comment]
@MrAnno Thanks, added it for every clang job, if it is okay.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by alltilla** [comment]
@bazsi: I received this info from the team.
Anyways, the problem is, that `pylint` fails on it. I tried to fix the linter errors, but stopped at here, where I couldn't decide how it should be fixed, and did not want to dig deep into the source:
```
modules/python/pylib/syslogng/debuggercli/commandlinelexer.py:69: [E1111(assignment-from-no-return), CommandLineLexer._get_next_token] Assigning result of a function call, where the function has no return
modules/python/pylib/syslogng/debuggercli/commandlinelexer.py:71: [E1111(assignment-from-no-return), CommandLineLexer._get_next_token] Assigning result of a function call, where the function has no return
```
At the meeting yesterday we concluded, that we can leave the `pylint` test in the CI with accepting failure, but it should be fixed some day.

File: CMakeLists.txt [hunk 2]
```diff
@@ -364,7 +364,8 @@ if (ENABLE_EXTRA_WARNINGS)
     -Wmissing-format-attribute
     -Wold-style-definition
     -Wundef
-    -Wignored-qualifiers)
+    -Wignored-qualifiers
+    -Wno-misleading-indentation)
```
[the remark this answers is not in the document]
Comment (alltilla) [reply CMakeLists.txt:368]: Thanks, I will look into this Friday :)

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (alltilla) [reply CMakeLists.txt]: Thanks, it seems to work with the necessary version checks :) Fixed!

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**